packages/fx-core/src/plugins/resource/function/plugin.ts

```diff
@@ -540,15 +543,37 @@ export class FunctionPluginImpl {
       FunctionConfigKey.credential
     );
 
-    // Retrieve and do cleanup
-    const site = this.checkAndGet(this.config.site, FunctionConfigKey.site);
-    this.config.site = undefined;
-
     const webSiteManagementClient: WebSiteManagementClient = await runWithErrorCatchAndThrow(
       new InitAzureSDKError(),
       () => AzureClientFactory.getWebSiteManagementClient(credential, subscriptionId)
     );
 
+    let site: Site;
+    if (isArmSupportEnabled()) {
```

Comment:
I think the post provision flow is becoming more and more complex. Both in frontend and function plugin. I really suggest to split the arm flow and origin flow to make things easier.

Reply:
OK. I split them. Thanks.